File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    var answer = [0, 0];
    const xRange = (board[0] - 1) / 2;
    const yRange = (board[1] - 1) / 2;
    keyinput.forEach((value) => {
        switch(value){
            case 'left':
                if(answer[0] > -xRange){
                    answer[0] -= 1;
                }
                break;
            case 'right':
                if(answer[0] < xRange){
                    answer[0] += 1;
                }
                break;
            case 'up':
                if(answer[1] < yRange){
                    answer[1] += 1;
                }
                break;   
            case 'down':
                if(answer[1] > -yRange){
                    answer[1] -= 1;
                }
                break;
            
        }
    })
    return answer;
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    x, y = 0, 0
    r_x = board[0] // 2
    r_y = board[1] // 2
    
    for k in keyinput:
        if k == "up" and y < r_y:
            y += 1
        elif k == "down" and y > -r_y:
            y -= 1
        elif k == "left" and x > -r_x:
            x -= 1
        elif k == "right" and x < r_x:
            x += 1
    return [x, y]